iroha/src/genesis.rs

```diff
@@ -0,0 +1,110 @@
+//! This module contains execution Genesis Block logic, and GenesisBlock definition.
+use crate::{init::config::InitConfiguration, tx::Accept, tx::AcceptedTransaction, Identifiable};
+use iroha_crypto::KeyPair;
+use iroha_data_model::{account::Account, isi::InstructionBox, prelude::*};
+use serde::Deserialize;
+use std::{fmt::Debug, fs::File, io::BufReader, path::Path};
+
+/// GenesisBlock contains transactions for inisialize settings.
+#[derive(Clone, Debug)]
+pub struct GenesisBlock {
+    /// transactions from GenesisBlock, any transaction is accepted
+    pub txs: Vec<AcceptedTransaction>,
+}
+
+#[derive(Clone, Deserialize, Debug)]
+struct RawGenesisBlock {
+    pub txs: Vec<GenesisTx>,
+}
+
+/// GenesisTx is a transaction for inisialize settings.
+#[derive(Clone, Deserialize, Debug)]
+struct GenesisTx {
+    isi: Vec<InstructionBox>,
+}
+
+impl GenesisBlock {
+    /// Construct `GenesisBlock` from `GenesisConfiguration`.
+    pub fn from_configuration(
+        config: &config::GenesisConfiguration,
+        init_config: &InitConfiguration,
+    ) -> Result<GenesisBlock, String> {
+        Ok(
+            GenesisBlock::load_genesis_file_from_path(&config.genesis_file_path, &init_config)
+                .expect("Failed to open a genesis block file"),
+        )
+    }
+
+    /// This method will build `Configuration` from a json *pretty* formatted file (without `:` in
+    /// key names).
+    /// # Panics
+    /// This method will panic if configuration file presented, but has incorrect scheme or format.
+    /// # Errors
+    /// This method will return error if system will fail to find a file or read it's content.
+    fn load_genesis_file_from_path<P: AsRef<Path> + Debug>(
+        path: P,
+        init_config: &InitConfiguration,
+    ) -> Result<GenesisBlock, String> {
+        let file =
+            File::open(path).map_err(|e| format!("Failed to open a genesis block file: {}", e))?;
+        let reader = BufReader::new(file);
+        let raw_block: RawGenesisBlock = serde_json::from_reader(reader)
+            .map_err(|e| format!("Failed to deserialize json from reader: {}", e))?;
+        let genesis_key_pair = KeyPair {
```

Comment:
Should I make constructor of KeyPair?

Reply:
No I don't think we need constructor for the `KeyPair`, it would simply fill the fields...

Reply:
ok!